### ISSUE TITLE
Fix Request for HttpClientBuilderPoolSettingsIgnored false positive 

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#192]
HttpAsyncClientBuilder with callbacks is not supported yet.
Example:
```
class DataStoreConfigGood1 {
    private RestClientBuilder getRestClientBuilder() {
        return RestClient.builder(hosts)
                .setHttpClientConfigCallback(this::createHttpClientConfigCallback);
    }

    private HttpAsyncClientBuilder createHttpClientConfigCallback(final HttpAsyncClientBuilder clientBuilder) {
        clientBuilder
                .setMaxConnPerRoute(20)
                .setMaxConnTotal(20);
        return clientBuilder;
    }
}
class DataStoreConfigGood2 {
    private RestClientBuilder getRestClientBuilder() {
        return RestClient.builder(hosts)
                .setHttpClientConfigCallback(this::createHttpClientConfigCallback);
    }

    private HttpAsyncClientBuilder createHttpClientConfigCallback(final HttpAsyncClientBuilder clientBuilder) {
        clientBuilder.setConnectionManager(connMgr);
        return clientBuilder;
    }
}
class DataStoreConfigBad {
    private RestClientBuilder getRestClientBuilder() {
        return RestClient.builder(hosts)
                .setHttpClientConfigCallback(this::createHttpClientConfigCallback);
    }

    private HttpAsyncClientBuilder createHttpClientConfigCallback(final HttpAsyncClientBuilder clientBuilder) {
        clientBuilder
                .setConnectionManager(connMgr)
                .setMaxConnPerRoute(20) // bad
                .setMaxConnTotal(20); // bad
        return clientBuilder;
    }
}
```
